identity/src/keypair.rs

```diff
@@ -300,6 +372,66 @@ impl From<rsa::Keypair> for Keypair {
     }
 }
 
+#[cfg(feature = "ed25519")]
+impl TryInto<ed25519::Keypair> for Keypair {
+    type Error = OtherVariantError;
+
+    fn try_into(self) -> Result<ed25519::Keypair, Self::Error> {
+        #[allow(deprecated)]
+        match self {
+            Keypair::Ed25519(inner) => Ok(inner),
+            Keypair::Rsa(_) => Err(OtherVariantError::new(KeyType::RSA)),
+            Keypair::Secp256k1(_) => Err(OtherVariantError::new(KeyType::Secp256k1)),
+            Keypair::Ecdsa(_) => Err(OtherVariantError::new(KeyType::Ecdsa)),
```

Comment:
These are missing `cfg` blocks for each branch. You'll need to check that all of the following commands succeed:

- `cargo check -p libp2p-identity --features ed25519 --all-targets`
- `cargo check -p libp2p-identity --features rsa --all-targets`
- `cargo check -p libp2p-identity --features secp256k1 --all-targets`
- `cargo check -p libp2p-identity --features ecdsa --all-targets`